R/mrgsolve.R

```diff
@@ -489,6 +547,10 @@ do_mrgsim <- function(x,
     x <- update_outputs(x, outputs = Request)  
   }
   
+  if(!is.character(etasrc) && length(etasrc)==1) {
```

Comment:
If I understand correctly, this is meant to catch anything that's not a character scalar.  So I guess the condition should be either `!(is.character(etasrc) && length(etasrc) == 1)` or `!is.character(etasrc) || length(etasrc) != 1`?

Examples of cases that I think should be caught but aren't, leading to an error message that's a little less helpful:

```
> mrgsim(mod, data, etasrc = c("data","foo"))
Error: Expecting a single string value: [type=character; extent=2].
> mrgsim(mod, data, etasrc = 1:2)
Error: Expecting a single string value: [type=integer; extent=2].
```

Reply:
Thanks, @kyleam. Yes ... the logic I had wasn't right. Will update to your first proposal.

Reply:
Both of these are passing now

``` r
library(mrgsolve)
#> 
#> Attaching package: 'mrgsolve'
#> The following object is masked from 'package:stats':
#> 
#>     filter
code <- '
$OMEGA
1 1 1 1 1 1 1 1 1 1
1 1 1 1 1 1 1 1 1 1

$CAPTURE ETA(1), ETA(9), ETA(10), ETA(16), ETA(20), ETA(21)
'

mod <- mcode("code", code)
#> Building code ...
#> done.


data <- expand.ev(
  amt = 100, ETA1 = 2, ETA5 = 5, ETA9 = 9, ET10 = 10, ET16 = 16, 
  ETA20 = 20, ET21 = 21, cmt = 0
)

mrgsim(mod, data, etasrc = c("data", "foo"))
#> Error:
#> ! `etasrc` must be a string.

#> Backtrace:
#>     ▆
#>  1. └─mrgsolve::mrgsim(mod, data, etasrc = c("data", "foo"))
#>  2.   └─mrgsolve::mrgsim_d(x, data = data, ...) at m4solve/R/mrgsolve.R:272:6
#>  3.     ├─base::do.call(...) at m4solve/R/mrgsolve.R:349:2
#>  4.     └─mrgsolve (local) `<fn>`(...)
#>  5.       └─rlang::abort("`etasrc` must be a string.") at m4solve/R/mrgsolve.R:551:4

mrgsim(mod, data, etasrc = c(1,2))
#> Error:
#> ! `etasrc` must be a string.

#> Backtrace:
#>     ▆
#>  1. └─mrgsolve::mrgsim(mod, data, etasrc = c(1, 2))
#>  2.   └─mrgsolve::mrgsim_d(x, data = data, ...) at m4solve/R/mrgsolve.R:272:6
#>  3.     ├─base::do.call(...) at m4solve/R/mrgsolve.R:349:2
#>  4.     └─mrgsolve (local) `<fn>`(...)
#>  5.       └─rlang::abort("`etasrc` must be a string.") at m4solve/R/mrgsolve.R:551:4
```

<sup>Created on 2023-02-21 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>